capstone/src/test/java/com/google/sps/servlets/CommentTest.java

```diff
@@ -0,0 +1,92 @@
+package com.google.sps.objects;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.gson.Gson;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.Objects.Comment;
+
+/**
+ * Unit tests for Comment.
+ *
+ */
+public class CommentTest {
+
+  private static final String USER_ID = "123123123";
+  private static final String COMMENT_TEXT = "a great comment";
+  private static final long TIMESTAMP = 4324344;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig()
+              .setDefaultHighRepJobPolicyUnappliedJobPercentage(0));
+
+  private Comment comment;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    comment = 
+      new Comment(
+        TIMESTAMP, /* timestamp */ 
+        COMMENT_TEXT, /* commentText */ 
+        USER_ID /* userId */ );
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    comment = null;
+  }
+
+  @Test
+  public void getCommentTextTest() {
+    assertEquals(comment.getCommentText(), COMMENT_TEXT);
+  }
+
+  @Test
+  public void getUserTest() {
+    assertEquals(comment.getUser(), USER_ID);
+  }
+
+  @Test
+  public void getCommentEntityTest() {
+    EmbeddedEntity entity = new EmbeddedEntity();
+    entity.setProperty("userId", USER_ID);
+    entity.setProperty("commentText", COMMENT_TEXT);
+    entity.setProperty("timestamp", TIMESTAMP);
+    Comment returnedComment = Comment.getCommentEntity(entity);
+
+    String jsonRetrieved = new Gson().toJson(returnedComment);
```

Comment:
Since you're comparing the json here is there a reason to have that equals method in Comment.java?

Reply:
I would only point out that the tests should not define your production API's. But if nobody is actually USING the `equals` functions, then you all certainly don't have to define them.

Reply:
I'm using the comment equals function in PostTest, but can also change it to a json comparison to keep things consistent.